ZenPacks/zenoss/Import4/bin/imp4_check.sh

```diff
@@ -53,56 +61,87 @@ do
+    ! gunzip -vf "zep.sql.gz" >&2 && err_exit "Invalid zep.sql.gz! abort"
+else
+    info_out "No Zeneventserver indexes, continue"
+fi
 
-info_out "Extracting zep.tar..."
-tar -vxf zep.tar 2>/dev/null | awk "$awk_cmd" >&2
-[[ ${PIPESTATUS[0]} -ne 0 ]] && info_out "Zeneventserver indexes not found, skipping"
+if [ -f zep.tar ]
+then
+    status_out "Extracting zep.tar"
+    tar -vxf zep.tar 2>/dev/null | awk "$awk_cmd" >&2
+    [[ ${PIPESTATUS[0]} -ne 0 ]] && err_exit "Extracting Zeneventserver indexes failed! abort"
+else
+    info_out "No Zeneventserver indexes archive, continue"
+fi
 
-# make sure dmd_uuid.txt is there!
-if [[ ! -f ../dmd_uuid.txt ]]; then
-    err_exit "dmd_uuid.txt is missing from backup, cannot continue"
+# catalogservice file is optional
+if [ -f zencatalogservice.tar ]
+then
+    status_out "Extracting zencatalogservice.tar"
+    tar -vxf zencatalogservice.tar | awk "$awk_cmd" >&2
+    [[ ${PIPESTATUS[0]} -ne 0 ]] && err_exit "Extracting zencatalogservice.tar failed! abort"
+else
+    info_out "No catalogservice archive, continue"
 fi
 
-# extract the perf into the shared staging volume
+# prep the staging area
 ! mkdir -p "$staging_zenbackup_dir" && err_exit "Cannot create staging directory in the containter!"
 
-info_out "Extracting performance data tends to take a long time"
-tar -C "$staging_zenbackup_dir" -vxf perf.tar | awk "$awk_cmd" >&2
-[[ ${PIPESTATUS[0]} -ne 0 ]] && err_exit "Extracting performance data from perf.tar failed!"
+if [ -f perf.tar ]
+then
+    # extract the perf into the shared staging volume
+    status_out "Extracting perf.tar"
+    info_out "This operation tends to take a long time ..."
+    tar -C "$staging_zenbackup_dir" -vxf perf.tar | awk "$awk_cmd" >&2
+    [[ ${PIPESTATUS[0]} -ne 0 ]] && err_exit "Extracting performance data from perf.tar failed!"
+else
+    info_out "No performance data archive, continue"
+fi
 
 info_out "Setup proper access right"
-(find . -type d -exec chmod a+rwx {} \; >&2) &
-(find . -type f -exec chmod a+rw {} \; >&2) &
-(find "$staging_dir" -type d -exec chmod a+rwx {} \; >&2) &
-(find "$staging_dir" -type f -exec chmod a+rw {} \; >&2) &
+find . -type d -exec chmod a+rwx {} \; >&2 &
+find . -type f -exec chmod a+rw {} \; >&2 &
+find "$staging_dir" -type d -exec chmod a+rwx {} \; >&2 &
+find "$staging_dir" -type f -exec chmod a+rw {} \; >&2 &
 wait
 
 # use the mounted directory as the current directory
 # no redirect of stdout where meta data is reported
 cmd="cd /mnt/pwd; /opt/zenoss/bin/python /import4/pkg/bin/import4 "
+status_out "Finding import meta data"
 ! su - zenoss -c "$cmd model check"            && err_exit "Model files not valid!"
-! su - zenoss -c "$cmd events check"           && err_exit "Events files not valid!"
-! su - zenoss -c "$cmd perf --skip-scan check" && err_exit "Performance data files not valid!"
+
+# optional prevalidation
+[ -f zep.sql ]  && ! su - zenoss -c "$cmd events check"           && err_exit "Events files not valid!"
```

Comment:
Yes, you are right. I now moved the zep indexes to follow the successful unzip of zep.sql.gz.

However, for zep.sql, it will only exist if the unzip of zep.sql.gz is successful. It is a check of the successful unzip, not the existance of the tar. So, line 143 should be fine.